tools/fiptool/fiptool.c

```diff
@@ -52,34 +52,6 @@
 #define OPT_PLAT_TOC_FLAGS 1
 #define OPT_ALIGN 2
 
-static image_desc_t *lookup_image_desc_from_uuid(const uuid_t *uuid);
```

Comment:
The "cmds" is prefixed with "static", so it is no more global than any other static functions.

The "cmds" is an array of function callbacks.
The callback functions are usually defined above the ops array to save declarations.
We see the same pattern everywhere in drivers.

    static int my_foo_read(...)
    {
           ...
    }

    static int my_foo_write(...)
    {
            ...
    }

    static struct foo_driver_ops my_foo_ops = {
            .read = my_foo_read,
            .write = my_foo_write,
            ....
    };
 
If you look at Linux Kernel, drivers are like that.  Callback functions first, then operation structure that contains them.

If you want to see some examples in ATF,
"git grep -C10 DECLARE_RT_SVC".

Of course, it is possible to put the DECLARE_RT_SVC() at the top of file if you like
(you need to add some declarations at the very top), what does it buy you?